### PR TITLE
Fix navigation in table by 'tab' press

### PR DESCRIPTION
**Pull request by Krijovnick**

### CODE REVIEW

File: modules/keyboard.js [hunk 1]
```diff
@@ -451,8 +451,9 @@ Keyboard.DEFAULTS = {
     },
     tab: {
       key: 'tab',
-      handler(range, context) {
-        if (context.format.table) return true;
+      handler(range, { format }) {
+        if (format.tableCellLine || format.tableHeaderCellLine
```
Comment (ksercs) [inline modules/keyboard.js:455]: I'd like to make this return more explicit
```javascript
const isInTable = format.tableCellLine || format.tableHeaderCellLine
    || format.tableHeaderCell || format.table;
if (isInTable) {
    return true;
}

```

File: test/unit/modules/keyboard.js [hunk 2]
```diff
@@ -257,4 +260,245 @@ describe('Keyboard', function () {
...
+      const keydownEvent = new KeyboardEvent('keydown', {
+        key: 'tab',
+        shiftKey: true,
+      });
+
+      quill.root.dispatchEvent(keydownEvent);
+      expect(quill.getSelection()).toEqual(new Range(29));
+    });
+
+    it('should select previous cell on first row if cursor in the first cell of second row on tab + shift click', function () {
```
Comment (ksercs) [inline test/unit/modules/keyboard.js:335]: just a little bit concretised
```suggestion
    it('should select last cell of the first row by tab+shift if cursor is in the first cell of the second row', function () {
```

File: test/unit/modules/keyboard.js [hunk 3]
```diff
@@ -257,4 +260,245 @@ describe('Keyboard', function () {
...
+      });
+
+      quill.root.dispatchEvent(keydownEvent);
+      expect(quill.getSelection()).toEqual(new Range(5));
+    });
+  });
+
+  describe('tab navigation on lite table', function () {
+    beforeAll(function () {
+      Quill.register({ 'modules/table': TableLite }, true);
```
Comment (ksercs) [inline test/unit/modules/keyboard.js:385]: Cannot this code be executed before other test? In this case all tests will be run for lite table. Can we add restore?

File: test/unit/modules/keyboard.js [hunk 4]
```diff
@@ -257,4 +260,245 @@ describe('Keyboard', function () {
...
+      quill.setSelection(18);
+      const keydownEvent = new KeyboardEvent('keydown', {
+        key: 'tab',
+      });
+
+      quill.root.dispatchEvent(keydownEvent);
+      expect(quill.getSelection()).toEqual(new Range(24));
+    });
+
+    it('should select next cell in second row if cursor in the last cell of first row on tab click', function () {
```
Comment (ksercs) [inline test/unit/modules/keyboard.js:426]:
```suggestion
    it('should select the first cell in the by tab press if cursor is in the last cell of the first row', function () {
```

File: test/unit/modules/keyboard.js [hunk 5]
```diff
@@ -257,4 +260,245 @@ describe('Keyboard', function () {
...
+      const keydownEvent = new KeyboardEvent('keydown', {
+        key: 'tab',
+        shiftKey: true,
+      });
+
+      quill.root.dispatchEvent(keydownEvent);
+      expect(quill.getSelection()).toEqual(new Range(29));
+    });
+
+    it('should select previous cell on first row if cursor in the first cell of second row on tab + shift click', function () {
```
Comment (ksercs) [inline test/unit/modules/keyboard.js:457]:
```suggestion
    it('should select the last cell of the first row by tab+shift if cursor us in the first cell of the second row, function () {
```

File: test/unit/modules/keyboard.js (hunk 3 above)
Comment (Krijovnick) [reply test/unit/modules/keyboard.js:386]: fixed

**Review by marker-dao — APPROVED** [review]

**Pull request merged**